crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -0,0 +1,1426 @@
+        Ok(self
+            .query_row(
+                "SELECT data FROM media WHERE uri = ? AND format = ?",
+                (uri, format),
+                |row| row.get(0),
+            )
+            .await
+            .optional()?)
+    }
+
+    async fn remove_media(&self, uri: Key, format: Key) -> Result<()> {
+        self.execute("DELETE FROM media WHERE uri = ? AND format = ?", (uri, format)).await?;
+        Ok(())
+    }
+
+    async fn remove_uri_medias(&self, uri: Key) -> Result<()> {
+        self.execute("DELETE FROM media WHERE uri = ?", (uri,)).await?;
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl SqliteObjectStateStoreExt for deadpool_sqlite::Object {
+    async fn set_kv_blob(&self, key: Key, value: Vec<u8>) -> Result<()> {
+        Ok(self.interact(move |conn| conn.set_kv_blob(&key, &value)).await.unwrap()?)
+    }
+}
+
+#[async_trait]
+impl StateStore for SqliteStateStore {
+    type Error = Error;
+
+    async fn get_kv_data(&self, key: StateStoreDataKey<'_>) -> Result<Option<StateStoreDataValue>> {
+        self.acquire()
+            .await?
+            .get_kv_blob(self.encode_state_store_data_key(key))
+            .await?
+            .map(|data| {
+                let string = self.deserialize_value(&data)?;
+                Ok(match key {
+                    StateStoreDataKey::SyncToken => StateStoreDataValue::SyncToken(string),
+                    StateStoreDataKey::Filter(_) => StateStoreDataValue::Filter(string),
+                    StateStoreDataKey::UserAvatarUrl(_) => {
+                        StateStoreDataValue::UserAvatarUrl(string)
+                    }
+                })
+            })
+            .transpose()
+    }
+
+    async fn set_kv_data(
+        &self,
+        key: StateStoreDataKey<'_>,
+        value: StateStoreDataValue,
+    ) -> Result<()> {
+        let value = match key {
+            StateStoreDataKey::SyncToken => {
+                value.into_sync_token().expect("Session data not a sync token")
+            }
+            StateStoreDataKey::Filter(_) => value.into_filter().expect("Session data not a filter"),
+            StateStoreDataKey::UserAvatarUrl(_) => {
+                value.into_user_avatar_url().expect("Session data not an user avatar url")
+            }
+        };
+
+        self.acquire()
+            .await?
+            .set_kv_blob(self.encode_state_store_data_key(key), self.serialize_value(&value)?)
+            .await
+    }
+
+    async fn remove_kv_data(&self, key: StateStoreDataKey<'_>) -> Result<()> {
+        self.acquire().await?.delete_kv_blob(self.encode_state_store_data_key(key)).await
+    }
+
+    async fn save_changes(&self, changes: &StateChanges) -> Result<()> {
+        let changes = changes.to_owned();
+        let this = self.clone();
+        self.acquire()
+            .await?
+            .with_transaction(move |txn| {
+                let StateChanges {
+                    sync_token,
+                    session: _,
```

Comment:
Do the other stores also ignore this one?

Reply:
Yes, I'm not event sure why it's in StateChanges.